project/apps/questions-service/src/questions.service.ts

```diff
@@ -80,14 +117,25 @@ export class QuestionsService {
   }
 
   async update(question: UpdateQuestionDto): Promise<QuestionDto> {
-    const updatedQuestion = {
-      ...question,
-      updated_at: new Date(),
-    };
+    const { data: existingQuestion } = await this.supabase
+      .from(this.QUESTIONS_TABLE)
+      .select()
+      .eq('q_title', question.q_title)
+      .neq('id', question.id)
+      .single<QuestionDto>();
```

Comment:
should there also be a check that we arent updating a deleted question? unless we plan to allow updating from deleted to not-deleted

Reply:
hm makes sense to add a check that `deleted_at === null`. 

don't think we should piggy-back the un-deleting of questions on this endpoint, that should be handled by another specific route like `PUT /questions/:id/restore` maybe